napari/layers/base/base.py

```diff
@@ -999,6 +1009,131 @@ def _data_to_world(self) -> Affine:
         """
         return self._transforms[1:3].simplified
 
+    def vector_world_to_data(self, vector) -> tuple:
```

Comment:
Maybe we use the word ray instead of vector? while i agree vector is reasonable, it also might make folks think of the vector layer, and I think ray is quite nice. We could use it everywhere.

Also how about making this method private, say `_world_to_data_ray`, and then adding a kwarg to our existing method

```python
def world_to_data(position, ray=False):
```

I know @jni might be a bit uneasy about a kwarg changing the meaning of the input, so maybe i should be overruled here, but just trying to think about the best naming. As noted in another comment below, you could do the same thing for `get_value(position, ray=True)` which would be neat!

Reply:
Hmm I'm a bit torn on this one. My current leaning is towards having a separate method (we can choose a different name) for both transforming orientations (`vector_world_to_data()`) and getting a value from a click since I think they are doing different things. For example, for `get_value`, in 2D, the click point really is the point in the data the user intends to interrogate. Whereas, in 3D, the click position is actually a position in a plane unknown to the user in the data and there are different ways we can interrogate the data using that point.

Reply:
Ok, let's see what @jni thinks, but if we stick with two methods i might prefer `world_to_day_ray` as a name and using `ray` as our word throughout

Reply:
I might be misunderstanding, but I'm actually kinda leaning towards unifying the API. It feels like the same principle — get a 2D point in canvas, then depending on `ndisplay`, this point implies either a 2D point in the data, or a 3D ray in the data.

In other words, we can use not a keyword argument like `ray=True`, but the dimensionality of `ray`/display as the switch.

btw the docstring below seems out of date, as `dims_displayed` is not an argument of this method...